pkg/registry/porch/packagerevision.go

```diff
@@ -166,6 +173,28 @@ func (r *packageRevisions) Create(ctx context.Context, runtimeObject runtime.Obj
 		parentPackage = p
 	}
 
+	pkgMutexKey := fmt.Sprintf("%s-%s-%s-%s",
```

Comment:
At this point in the flow, the PackageRevision's name (as seen in kubectl) hasn't yet been generated - that's done a good bit later after creating a PackageDraft and Close()'ing it to get a repoPkgRev whose KubeObjectName() method generates the final PackageRevision name by check-summing from repo, package path, and workspace name (and the namespace makes it unique across namespaces). Rather than recreate that process redundantly, I strung the four parts together here to use as the mutex key - it's not more detailed, really, just that the parts are put together differently.